lib/vm/opFns.js

```diff
@@ -91,7 +90,16 @@ module.exports = {
     if (exponent.isZero()) {
       return new BN(1)
     } else {
-      var bytes = 1 + logTable(exponent)
+      let bytesAdd = 32
+      for (let i = 1; i <= 32; i++) {
+        // e.g. 1 -> 0100, 8 -> 010000000000000000
+        let expCmp = new BN('01' + new Array(i * 2).fill(0).join(''), 16)
+        if (exponent.cmp(expCmp) === -1) {
+          bytesAdd = i - 1
+          break
+        }
+      }
+      var bytes = 1 + bytesAdd
```

Comment:
Just pushed. Totally insane, sorry, I always need some time till I get the whole picture, yes that was just the extracted functionality. 😄 😛 😄

Reply:
I also cheated from [py-evm](https://github.com/ethereum/py-evm/blob/master/eth/vm/logic/arithmetic.py#L150) :sweat_smile: 

Just occured to me, maybe `1 <= byteLength <= 32` might be a helpful safety check. This I think is the downside of `BN`, that numbers can be arbitrarily large. We should at some point maybe consider something like [fixed-bn.js](https://github.com/ewasm/fixed-bn.js).

Reply:
Ok, have added a check, let me know if you are not happy with the naming of the error type I added.

Side note: I actually have no idea of the purpose of this extracted ``trap`` function, wondered several times already. For me this would be more readable if one would directly throw in-place, not sure what's your stance on this.

Reply:
The name makes sense IMO, only there are two linting errors.

I think one argument for `trap` is that if we throw in each occasion, the errors might become inconsistent (trap always throws a `VmError`). But overall I'm okay both ways.

Reply:
Have fixed the linting.